esm-wrapper.mjs

```diff
@@ -0,0 +1,145 @@
+// just pre-load all the stuff that index.js lazily exports
+import {createRequire} from 'module';
+
+const require = createRequire(import.meta.url);
+
+const internalRe = require('./internal/re');
+
+const re = internalRe.re;
+const src = internalRe.src;
+const tokens = internalRe.t;
+const SEMVER_SPEC_VERSION = require('./internal/constants').SEMVER_SPEC_VERSION;
+const SemVer = require('./classes/semver');
+const compareIdentifiers = require('./internal/identifiers').compareIdentifiers;
+const rcompareIdentifiers = require('./internal/identifiers').rcompareIdentifiers;
+const parse = require('./functions/parse');
+const valid = require('./functions/valid');
+const clean = require('./functions/clean');
+const inc = require('./functions/inc');
+const diff = require('./functions/diff');
+const major = require('./functions/major');
+const minor = require('./functions/minor');
+const patch = require('./functions/patch');
+const prerelease = require('./functions/prerelease');
+const compare = require('./functions/compare');
+const rcompare = require('./functions/rcompare');
+const compareLoose = require('./functions/compare-loose');
+const compareBuild = require('./functions/compare-build');
+const sort = require('./functions/sort');
+const rsort = require('./functions/rsort');
+const gt = require('./functions/gt');
+const lt = require('./functions/lt');
+const eq = require('./functions/eq');
+const neq = require('./functions/neq');
+const gte = require('./functions/gte');
+const lte = require('./functions/lte');
+const cmp = require('./functions/cmp');
+const coerce = require('./functions/coerce');
+const Comparator = require('./classes/comparator');
+const Range = require('./classes/range');
+const satisfies = require('./functions/satisfies');
+const toComparators = require('./ranges/to-comparators');
+const maxSatisfying = require('./ranges/max-satisfying');
+const minSatisfying = require('./ranges/min-satisfying');
+const minVersion = require('./ranges/min-version');
+const validRange = require('./ranges/valid');
+const outside = require('./ranges/outside');
+const gtr = require('./ranges/gtr');
+const ltr = require('./ranges/ltr');
+const intersects = require('./ranges/intersects');
+const simplifyRange = require('./ranges/simplify');
+const subset = require('./ranges/subset');
+
+export default {
```

Comment:
unsure if you want to support the default or not... we could remove this if you want to keep the interface limited

Reply:
The default export is required for this to not be semver-major, otherwise code that already uses `import semver from 'semver'` will break.

Side note the approach I've taken for semver-minor ESM wrappers is as follows.  @guybedford do you see issues with this for bundlers?
```js
import semver from './index.js'

export const {/* list all named exports here */} = semver

export default semver
```

Then you can test with something like:
```js
// test/esm/test-exports.mjs
import {createRequire} from 'module'
import t from 'tap'

import * as semver from 'semver'

const require = createRequire(import.meta.url)

const cjs = require('semver')

t.equal(cjs, semver.default)
t.same(cjs, Object.fromEntries(Object.entries(semver).filter(nvp => nvp[0] !== 'default')))
```

This verifies that `package.json#exports` is correct for require and import on the main path, then it verifies identical API among the conditional exports.

Reply:
So there is an interesting issue with semver... it installs a module which depends on semver, which results in it being in the node_modules folder... which breaks self-referential modules afaict

Reply:
Not when I tested my suggestion locally on node.js 14.1.0, `node_modules/semver` was ignored by the self referential import and require.